compiler/src/dotty/tools/dotc/transform/MoveStatics.scala

```diff
@@ -24,7 +24,8 @@ class MoveStatics extends MiniPhase with SymTransformer {
   override def phaseName: String = MoveStatics.name
 
   def transformSym(sym: SymDenotation)(implicit ctx: Context): SymDenotation = {
-    if (sym.hasAnnotation(defn.ScalaStaticAnnot) && sym.owner.is(Flags.Module) && sym.owner.companionClass.exists) {
+    if (sym.hasAnnotation(defn.ScalaStaticAnnot) && sym.owner.is(Flags.Module) &&
+        sym.owner.companionClass.exists && !sym.owner.companionClass.is(Flags.Trait)) {
```

Comment:
According to https://github.com/lampepfl/dotty/issues/5924#issuecomment-464079014 it's OK to do the move if the field is final so why not add:
```suggestion
        sym.owner.companionClass.exists && !sym.owner.companionClass.is(Flags.Trait, butNot = Final)) {
```

Furthermore, I think that the `OFFSET$` fields generated by the lazy val phase could be marked final in bytecode, so they could still be moved.

Reply:
Interestingly, `OFFSET$` is actually immutable but changed to mutable in MoveStatics:

https://github.com/lampepfl/dotty/blob/12eef6d8e51dcd8a9a73387659b0f860d7b9ca80/compiler/src/dotty/tools/dotc/transform/MoveStatics.scala#L30